evap/results/views.py

```diff
@@ -81,7 +85,11 @@ def course_detail(request, semester_id, course_id):
         results = []
         for result in section.results:
             if isinstance(result, TextResult):
-                answers = [answer for answer in result.answers if user_can_see_text_answer(request.user, represented_users, answer, public_view)]
+                if view == 'filter':
+                    answers = [answer for answer in result.answers if user_can_see_text_answer(UserProfile.objects.get(id=contributor_id), [], answer, view)]
```

Comment:
the `.get` executes a database query, which is rather expensive, so this should be done repeatedly in a loop.